tempest/api/workloadmgr/restore/test_exclude_cinder_volume.py

```diff
@@ -0,0 +1,266 @@
+
+            self.wait_for_snapshot_tobe_available(wid, snapshot_id)
+            if(self.getRestoreStatus(wid, snapshot_id, restore_id_1) == "available"):
+                reporting.add_test_step("Selective restore", tvaultconf.PASS)
+            else:
+                reporting.add_test_step("Selective restore", tvaultconf.FAIL)
+                raise Exception("Selective restore failed")
+
+            # Fetch instance details after restore
+            vm_list = self.get_restored_vm_list(restore_id_1)
+            LOG.debug("Restored vm(selective) ID : " + str(vm_list))
+
+            time.sleep(60)
+
+            #assign floating ip
+            floating_ip_2 = self.assign_floating_ips(vm_list[0], True)
+            LOG.debug(
+                "Floating ip assigned to selective restore vm -> " +
+                str(floating_ip_2))
+
+            #connect to restored vm and try to mount volume
+            max_retries = 0
+            cmd_err = ""
+            cmd_output = ""
+            ssh = self.SshRemoteMachineConnectionWithRSAKey(str(floating_ip_2))
+            cmd = "sudo mount "+volumes[0]+"1 "+mount_point[0]+" "
+            LOG.debug("Command to mount volume" + str(cmd))
+            stdin, stdout, stderr = ssh.exec_command(
+            cmd)
+            time.sleep(5)
+            while not stdout.channel.exit_status_ready():
+                time.sleep(3)
+                max_retries += 1
+                if max_retries == 10:
+                    reporting.add_test_step("Max retries exceeded to get command output.", tvaultconf.FAIL)
+                    raise Exception("Max retries exceeded to get command output.")
+
+            #read command output and error
+            cmd_err = stderr.readlines()
+            cmd_output = stdout.readlines()
+            LOG.debug("COMMAND error  " + str(cmd_err))
+            LOG.debug("COMMAND output " + str(cmd_output))
+
+            cmd_err = str(cmd_err)
+            if len(cmd_err) > 0:
+                err_msg = ["special device","does not exist"]
+                # look for all sub-string in error message.
+                result = all(x in cmd_err for x in err_msg)
+                if (result):
+                    reporting.add_test_step("Volume status post restore for metadata exclude_from_backup",
+                                            tvaultconf.PASS)
+                    reporting.set_test_script_status(tvaultconf.PASS)
+                    tests[1][1] = 1
+                else:
+                    reporting.add_test_step("Different error occurred than expected.", tvaultconf.FAIL)
+                    raise Exception("Different error occurred while performing mount operation.")
+            else:
+                reporting.add_test_step("Different error occurred than expected.", tvaultconf.FAIL)
+                raise Exception("Mount command successful, test case failed.")
+
+            #close ssh connection
+            ssh.close()
```

Comment:
In case of any exception, ssh connection does not get closed. Need to handle it.